homeassistant/components/nest/climate_sdm.py

```diff
@@ -319,16 +314,12 @@ async def async_set_preset_mode(self, preset_mode):
         """Set new target preset mode."""
         if preset_mode not in self.preset_modes:
             return
-        if ThermostatEcoTrait.NAME not in self._device.traits:
```

Comment:
Shouldn't this method throw a ValueError in case a preset is set that isn't valid?

Reply:
Yeah, I can see failing loudly in a "shouldn't happen" case makes more sense than failing silently.

I've changed the "not in preset_modes" check to raise a ValueError with a helpful error message.  I removed this second line of checking the device supported traits because it is not possible to test, so i'm hacking test coverage a bit to get above 96%, but at the same time it seems OK given the code that returns the preset_modes is doing this same check.

I updated tests to exercise this and expect the ValueError.